lib/src/main/kotlin/com/wire/sdk/service/EventsRouter.kt

```diff
@@ -228,20 +220,30 @@ internal class EventsRouter internal constructor(
             }
         }
 
-        groupId?.run {
-            when (wireEventsHandler) {
-                is WireEventsHandlerDefault -> wireEventsHandler.onAppAddedToConversation(
-                    conversation = conversationData,
-                    members = members
-                )
-                is WireEventsHandlerSuspending -> wireEventsHandler.onAppAddedToConversation(
-                    conversation = conversationData,
-                    members = members
-                )
-            }
+        val conversationModel = Conversation.fromDTO(conversationData)
+
+        when (wireEventsHandler) {
+            is WireEventsHandlerDefault -> wireEventsHandler.onAppAddedToConversation(
+                conversation = conversationModel,
+                members = members
+            )
+            is WireEventsHandlerSuspending -> wireEventsHandler.onAppAddedToConversation(
+                conversation = conversationModel,
+                members = members
+            )
         }
+    }
 
-        return mlsGroupId
+    private suspend fun handleRejoiningConversation(
+        qualifiedConversation: QualifiedId
+    ): MLSGroupId {
+        val conversationResponse = backendClient.getConversation(qualifiedConversation)
+        val conversation = conversationService.saveConversationWithMembers(
+            qualifiedConversation = qualifiedConversation,
+            conversationResponse = conversationResponse
+        ).first
+
+        return conversation.mlsGroupId
```

Comment:
the name is mismatching, it fetches conversation data and saves locally and returns the mlsGroupId, does no rejoining